services/src/main/java/org/keycloak/protocol/oid4vc/issuance/credentialoffer/InMemoryCredentialOfferStorageFactory.java

```diff
@@ -0,0 +1,31 @@
+package org.keycloak.protocol.oid4vc.issuance.credentialoffer;
+
+import org.keycloak.Config;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.KeycloakSessionFactory;
+
+public class InMemoryCredentialOfferStorageFactory implements CredentialOfferStorageFactory {
+
+    private static CredentialOfferStorage INSTANCE;
+
+    @Override
+    public CredentialOfferStorage create(KeycloakSession session) {
+        if (INSTANCE==null) {
```

Comment:
Would you mind auto-formatting the file?

```suggestion
        if (INSTANCE == null) {
```